src/main/java/com/efficio/fieldbook/web/common/service/impl/CrossingTemplateParser.java

```diff
@@ -149,37 +154,139 @@ private void parseObservationSheet(final String programUUID) throws FileParsingE
+					femaleStudyName);
+				cross.setFemaleParent(femaleParent);
+			} else {
+				throw new FileParsingException(this.messageSource.getMessage("no.list.data.for.plot",
+					new Object[] {femaleStudyName, crossInfo.getMiddle()}, LocaleContextHolder.getLocale()));
+			}
+
+			cross.setMaleParents(this.getMaleParents(maleNurseriesPlotMap, crossInfo));
+
 			final Germplasm germplasm = new Germplasm();
 			germplasm.setGnpgs(2);
 			germplasm.setGid(Integer.MAX_VALUE);
-			germplasm.setGpid1(femaleListDataProject.getGermplasmId());
-			germplasm.setGpid2(maleListDataProjects.get(0).getGermplasmId());
-			final String crossString = this.crossingService.getCross(germplasm, importedCrosses, "/");
-			importedCrosses.setCross(crossString);
+			germplasm.setGpid1(Integer.valueOf(cross.getFemaleGid()));
+			germplasm.setGpid2(cross.getMaleParents().get(0).getGid());
+			final String crossString = this.crossingService.getCross(germplasm, cross, "/");
+			cross.setCross(crossString);
+		}
+	}
 
-			this.importedCrossesList.addImportedCrosses(importedCrosses);
+	List<ImportedGermplasmParent> getMaleParents(
+		final Map<String, Map<Integer, ImportedCrossParent>> maleNurseriesPlotMap,
+		final Triple<String, Integer, List<Integer>> crossInfo) throws FileParsingException {
+		final String crossMaleStudy = crossInfo.getLeft();
+		final Map<Integer, ImportedCrossParent> malePlotMap = maleNurseriesPlotMap.get(crossMaleStudy);
+		final List<Integer> crossMalePlotNos = crossInfo.getRight();
+		final List<ImportedGermplasmParent> maleParents = new ArrayList<>();
+		for(final Integer crossMalePlotNo: crossMalePlotNos) {
+			if (malePlotMap != null && malePlotMap.containsKey(crossMalePlotNo)) {
+				final ImportedCrossParent maleCrossParent = malePlotMap.get(crossMalePlotNo);
+				final ImportedGermplasmParent maleParent = new ImportedGermplasmParent(maleCrossParent.getGid(),
+					maleCrossParent.getDesignation(), crossMalePlotNo, crossMaleStudy);
+				maleParents.add(maleParent);
+			} else if(crossMalePlotNo == 0 && crossMalePlotNos.size() == 1) {
+				final ImportedGermplasmParent maleParent = new ImportedGermplasmParent(crossMalePlotNo,
+					Name.UNKNOWN, crossMalePlotNo, crossMaleStudy);
+				maleParents.add(maleParent);
+			} else {
+				throw new FileParsingException(this.messageSource.getMessage("no.list.data.for.plot",
+					new Object[] {crossMaleStudy, crossMalePlotNo}, LocaleContextHolder.getLocale()));
+			}
+		}
+		return maleParents;
+	}
 
-			currentRow++;
+	/**
+	 * Returns map of plot numbers to corresponding ImportedCrossParent record from specified plot numbers
+	 * of given study
+	 *
+	 * @param studyName
+	 * @param plotNos
+	 * @param programUUID
+	 * @return
+	 * @throws FileParsingException
+	 */
+	Map<Integer, ImportedCrossParent> getPlotToImportedCrossParentMapForStudy(final String studyName, final Set<Integer> plotNos,
```

Comment:
There is no need to pass around `programUUID` as parameter through several methods from `parseWorkbook `method.
We can just use `this.contextUtil.getCurrentProgramUUID()` in the actual method where it's called which is `getPlotToImportedCrossParentMapForStudy`